tests/unit/migration/MigrationManagerBaseTests.cpp

```diff
@@ -0,0 +1,96 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "migration/MigratiorStatus.hpp"
+#include "migration/TestMigrators.hpp"
+#include "migration/impl/MigrationManagerBase.hpp"
+#include "migration/impl/MigratorsRegister.hpp"
+#include "util/MockMigrationBackend.hpp"
+#include "util/MockMigrationBackendFixture.hpp"
+#include "util/MockPrometheus.hpp"
+#include "util/config/Config.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <tuple>
+
+using TestMigratorRegister =
+    migration::impl::MigratorsRegister<MockMigrationBackend, SimpleTestMigrator, SimpleTestMigrator2>;
+
+using TestCassandraMigrationManager = migration::impl::MigrationManagerBase<TestMigratorRegister>;
+
+struct MigrationManagerBaseTest : public util::prometheus::WithMockPrometheus, public MockMigrationBackendTestStrict {
+    util::Config cfg;
+
+    std::shared_ptr<TestCassandraMigrationManager> migrationManager;
+
+    MigrationManagerBaseTest()
+    {
+        auto mockBackendPtr = backend.operator std::shared_ptr<MockMigrationBackend>();
+        TestMigratorRegister migratorRegister(mockBackendPtr);
+        migrationManager = std::make_shared<TestCassandraMigrationManager>(mockBackendPtr, cfg);
+    }
+};
+
+TEST_F(MigrationManagerBaseTest, AllStatus)
+{
+    EXPECT_CALL(*backend, fetchMigratorStatus("SimpleTestMigrator", testing::_)).WillOnce(testing::Return("Migrated"));
+    EXPECT_CALL(*backend, fetchMigratorStatus("SimpleTestMigrator2", testing::_))
+        .WillOnce(testing::Return("NotMigrated"));
+    auto const status = migrationManager->allMigratorsStatus();
+    EXPECT_EQ(status.size(), 2);
+    EXPECT_TRUE(
+        std::find(
+            status.begin(), status.end(), std::make_tuple("SimpleTestMigrator", migration::MigratorStatus::Migrated)
```

Comment:
Why `make_tuple`? I thought `Status` has a constructor.

Reply:
It is tuple of the name of the migrator and its status.

Reply:
So `status` here is a vector of tuples containing name and status. Maybe then it would be better to call it `migratorNameStatusPairs`.